### PR TITLE
[ONNX] Cap opset version at 17 for torch.onnx.export

### LINKED ISSUE #107446: [docs] [onnx] Update the max admissible `opset_version` in the docs of `torch.onnx.export` and maybe update the default `opset_version` to 17 in release 2.1?
### 🐛 Describe the bug

Nightly docs: https://pytorch.org/docs/master/onnx.html?highlight=export#torch.onnx.export
image

says that the max supported `opset_version` is 16, but searching through the issues one can see that https://github.com/pytorch/pytorch/issues/105134 even `opset_version = 18` does not error out and often `opset_version = 17` works

Is https://github.com/onnx/onnx/blob/main/docs/Operators.md#LayerNormalization supported by `torch.onnx.export`?

Updating the default `opset_version` to 17 can help to graph compilers, as more complex ops (like LayerNorm) are natively supported starting from `opset_version = 17`

### LINKED ISSUE #107801: [ONNX] cap support at opset 17
torch.onnx supports opset 17 max.

### PR DESCRIPTION
**Pull request by justinchuby**
Cap opset version at 17 for torch.onnx.export and suggest users to use the dynamo exporter. Warn users instead of failing hard because we should still allow users to create custom symbolic functions for opset>17.

Also updates the default opset version by running `tools/onnx/update_default_opset_version.py`.

Fixes #107801 Fixes #107446

### CODE REVIEW

File: torch/onnx/_constants.py [hunk 1]
```diff
@@ -5,8 +5,9 @@
 ONNX_BASE_OPSET = 9
 ONNX_MIN_OPSET = 7
 ONNX_MAX_OPSET = 18
+ONNX_OLD_EXPORTER_MAX_OPSET = 17
```
Comment (BowenBao) [inline torch/onnx/_constants.py:8]:
```suggestion
ONNX_TORCHSCRIPT_EXPORTER_MAX_OPSET = 17
```
nit

**Review by BowenBao — APPROVED** [review]
LG, is the unittest opset version also capped?

File: torch/onnx/_constants.py (hunk 1 above)
Comment (justinchuby) [reply torch/onnx/_constants.py:8]: Done

**Comment by justinchuby** [comment]
> LG, is the unittest opset version also capped?

Done

**Pull request closed without merging**